sam/cigar.go

```diff
@@ -197,6 +196,21 @@ func init() {
 	}
 }
 
+var powers = []int{1, 10, 100, 1000, 10000, 100000, 1000000, 10000000, 100000000}
+
+// atoi is an internal function; bytes in b are known to be digits.
+func atoi(b []byte) (int, error) {
+	n := 0
+	k := len(b) - 1
+	for i, v := range b {
+		n += int(v-'0') * powers[k-i]
+	}
+	if n >= 1<<28 || n < 0 {
+		return n, fmt.Errorf("sam: cigar-op size out of bounds [0, 1<<28): %d", n)
```

Comment:
`fmt.Errorf("sam: invalid cigar operation count: %q", b)`. Ideally, I'd like to have `i` here as well so the start can be pointed to in the error, so `fmt.Errorf("sam: invalid cigar operation count: %q at %d", b, i)`.